pkgs/development/tools/build-managers/gradle/default.nix

```
@@ -23,104 +23,133 @@ rec {
+            pname = "gradle";
+            inherit version;
+
+            src = fetchurl {
+              inherit sha256;
+              url =
+                "https://services.gradle.org/distributions/gradle-${version}-bin.zip";
+            };
+
+            dontBuild = true;
+
+            nativeBuildInputs = [ makeWrapper unzip ];
+            buildInputs = [ java ];
+
+            installPhase = with builtins;
+              let
+                javaHomeProperty = "org.gradle.java.home=" + java;
+                toolchainsProperty = "org.gradle.java.installations.paths=" + (concatStringsSep "," javaToolchains);
+                toolchainsPropertyWithCheck =
+                if (hasToolchainSupport || (length javaToolchains) == 0) then toolchainsProperty
+                else lib.warn "This Gradle version (${version}) does not have toolchain support, yet the 'javaToolchains' list is not empty. The used toolchain will be ignored by Gradle." toolchainsProperty;
+                propDefs = concatStringsSep "\n" [ javaHomeProperty toolchainsPropertyWithCheck ];
+              in
+              ''
+                mkdir -pv $out/lib/gradle/
+                cp -rv lib/ $out/lib/gradle/
+
+                gradle_launcher_jar=$(echo $out/lib/gradle/lib/gradle-launcher-*.jar)
+                test -f "$gradle_launcher_jar"
+                echo '${propDefs}' > $out/lib/gradle/gradle.properties
+                makeWrapper "${java}"/bin/java $out/bin/gradle \
+                  --add-flags "-classpath $gradle_launcher_jar org.gradle.launcher.GradleMain"
+              '';
+
+            dontFixup = !stdenv.isLinux;
+
+            fixupPhase =
+              let arch = if stdenv.is64bit then "amd64" else "i386";
+              in ''
+                for variant in "" "-ncurses5" "-ncurses6"; do
+                  mkdir "patching$variant"
+                  pushd "patching$variant"
+                  jar xf $out/lib/gradle/lib/native-platform-linux-${arch}$variant-${nativeVersion}.jar
+                  patchelf \
+                    --set-rpath "${stdenv.cc.cc.lib}/lib64:${lib.makeLibraryPath [ stdenv.cc.cc ncurses5 ncurses6 ]}" \
+                    net/rubygrapefruit/platform/linux-${arch}$variant/libnative-platform*.so
+                  jar cf native-platform-linux-${arch}$variant-${nativeVersion}.jar .
+                  mv native-platform-linux-${arch}$variant-${nativeVersion}.jar $out/lib/gradle/lib/
+                  popd
+                done
+
+                # The scanner doesn't pick up the runtime dependency in the jar.
+                # Manually add a reference where it will be found.
+                mkdir $out/nix-support
+                echo ${stdenv.cc.cc} > $out/nix-support/manual-runtime-dependencies
+                # Gradle will refuse to start without _both_ 5 and 6 versions of ncurses.
+                { echo "${ncurses5}"; echo "${ncurses6}"; } >> $out/nix-support/manual-runtime-dependencies
```

Slightly shorter even:

```suggestion
                echo -e "${ncurses5}\n${ncurses6}" >> $out/nix-support/manual-runtime-dependencies
```

---

The command before can also be merged

---

can you apply this change?